mesonbuild/mtest.py

```diff
@@ -1485,6 +1485,10 @@ def __init__(self, test: TestSerialisation, env: T.Dict[str, str], name: str,
         if ('MSAN_OPTIONS' not in env or not env['MSAN_OPTIONS']):
             env['MSAN_OPTIONS'] = 'halt_on_error=1:abort_on_error=1:print_summary=1:print_stacktrace=1'
 
+        # Valgrind also doesn't reflect errors in its exit code by default.
+        if ('VALGRIND_OPTS' not in env or not env['VALGRIND_OPTS']):
+            env['VALGRIND_OPTS'] = '--error-exitcode=1'
+
```

Comment:
Since we have access to `meson test` options here, maybe we should check whether there is a wrapper / whether the wrapper is valgrind?

Currently we set a bunch of variables and the resulting `meson test -v` output can be a bit overwhelming. I'm not sure there's a good solution to this, since we don't really know whether the test programs are built with sanitizers. But maybe it's easier to deduce this with valgrind?

I have mixed feelings about this since I suppose there's no reason someone couldn't run valgrind inside of `./run_tests.py`, instead of using `meson test --wrapper`. Do we want to offer enhanced handling for that?

Reply:
This is a fair comment and one I was anticipating, but I couldn't decide if I thought it was right or not. I agree that it's overwhelming and perhaps we should even go back and improve this for sanitizers another time too.

I'll look at doing it for Valgrind for now.

Reply:
Done.